beekeeper-api/src/test/java/com/expediagroup/beekeeper/api/HouseKeepingEntityServiceImplTest.java

```diff
@@ -0,0 +1,104 @@
+/**
+ * Copyright (C) 2019-2021 Expedia, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.expediagroup.beekeeper.api;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.mockito.Mockito.when;
+
+import static com.expediagroup.beekeeper.core.model.HousekeepingStatus.DELETED;
+import static com.expediagroup.beekeeper.core.model.HousekeepingStatus.SCHEDULED;
+import static com.expediagroup.beekeeper.core.model.LifecycleEventType.EXPIRED;
+
+import java.time.Duration;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.jpa.domain.Specification;
+
+import com.expediagroup.beekeeper.core.model.HousekeepingMetadata;
+import com.expediagroup.beekeeper.core.repository.HousekeepingMetadataRepository;
+
+@ExtendWith(MockitoExtension.class)
+public class HouseKeepingEntityServiceImplTest {
+
+  private HousekeepingMetadata table1;
+  private HousekeepingMetadata table2;
+  private HousekeepingEntityServiceImpl housekeepingEntityServiceImpl;
+
+  @Mock
+  private HousekeepingMetadataRepository housekeepingMetadataRepository;
+  @Mock
+  private Specification<HousekeepingMetadata> spec;
+  @Mock
+  private Pageable pageable;
+
+  @BeforeEach
+  public void createTables(){
+    housekeepingEntityServiceImpl = new HousekeepingEntityServiceImpl(housekeepingMetadataRepository);
+
+    LocalDateTime CREATION_TIMESTAMP = LocalDateTime.now(ZoneId.of("UTC"));
+
+    table1 = new HousekeepingMetadata.Builder()
```

Comment:
These also don't need to be generated in the beforeEach because all your tests won't require you to generate tables. Some will be testing errors and some will be returning empty lists. So its better to generate these tables inside your tests